frontend/components/RewardsHistory/RewardsHistory.tsx

```diff
@@ -0,0 +1,225 @@
+import {
+  ApiOutlined,
+  CloseOutlined,
+  HistoryOutlined,
+  InfoCircleOutlined,
+} from '@ant-design/icons';
+import {
+  Button,
+  Col,
+  ConfigProvider,
+  Flex,
+  Popover,
+  Row,
+  Spin,
+  Tag,
+  ThemeConfig,
+  Typography,
+} from 'antd';
+import { CSSProperties, ReactNode, useMemo } from 'react';
+import styled from 'styled-components';
+
+import { CardTitle } from '@/components/Card/CardTitle';
+import { CardFlex } from '@/components/styled/CardFlex';
+import { COLOR } from '@/constants/colors';
+import { UNICODE_SYMBOLS } from '@/constants/symbols';
+import { Pages } from '@/enums/PageState';
+import { usePageState } from '@/hooks/usePageState';
+import { balanceFormat } from '@/utils/numberFormatters';
+
+import { EpochDetails, StakingReward } from './types';
+import { useRewardsHistory } from './useRewardsHistory';
+
+const { Text, Title } = Typography;
+const MIN_HEIGHT = 400;
+const iconStyle: CSSProperties = { fontSize: 48, color: COLOR.TEXT_LIGHT };
+
+const yourWalletTheme: ThemeConfig = {
```

Comment:
just highlighting that this is repeating in YourWallet, maybe makes sense to move somewhere or apply globally (I know, might be harder)?